### PR TITLE
Feature/mico 588 add service to application should update version of services

### LINKED ISSUE #588: Endpoint `addServiceToApplication` should update version of included services
Only *MicoService* in one specific version is allowed per *MicoApplication*. Therefore we need a possibility to update an included service to a new version like described in #390.

Use the already existing endpoint `addServiceToApplication: POST /applications/{shortName}/{version}/services/{serviceShortName}/{serviceVersion}` for that.
Check if the service is already included and update the version if necessary.

### PR DESCRIPTION
**Pull request by wagnerdk**
- [X] Ensure that the commit message is [a good commit message](https://github.com/joelparkerhenderson/git_commit_message)
- [X] Ensure to use auto format in **all** files
- [X] Tests created for changes or:

---

## Short Description
Add the functionality to update the version of a service associated with a mico application.

## Resolving Issue/ Feature
Closes #588

### CODE REVIEW

**Comment by JakobBenz** [comment]
@Pascal157600 @JanVetter one more time, endpoint to add services and some private helper functions changed in `ApplicationResource`, corresponding checks, too.

File: mico-core/src/main/java/io/github/ust/mico/core/resource/ApplicationResource.java [hunk 1]
```diff
@@ -256,82 +256,72 @@
...
+                                                        @PathVariable(PATH_VARIABLE_VERSION) String applicationVersion,
+                                                        @PathVariable(PATH_VARIABLE_SERVICE_SHORT_NAME) String serviceShortName,
+                                                        @PathVariable(PATH_VARIABLE_SERVICE_VERSION) String serviceVersion) {
+    	// Retrieve application and service from database (checks whether they exist)
+        MicoApplication application = getApplicationFromDatabase(applicationShortName, applicationVersion);
+        MicoService service = getServiceFromDatabase(serviceShortName, serviceVersion);
+        
+        // Find all services with identical short name within this application
+		List<MicoService> micoServicesWithIdenticalShortName = serviceRepository
+		    .findAllByApplicationAndShortName(applicationShortName, applicationVersion, serviceShortName);
```
Comment (wagnerdk) [inline mico-core/src/main/java/io/github/ust/mico/core/resource/ApplicationResource.java]: Why did you change this? It is not necessary to query the information from the db again.  Everything needed is already there. We now get the same information twice from the db.

File: mico-core/src/main/java/io/github/ust/mico/core/resource/ApplicationResource.java [hunk 2]
```diff
@@ -256,82 +256,72 @@
...
-                    micoApplicationShortName, micoApplicationVersion, micoServiceShortName, micoServiceVersion, associatedMicService.getVersion());
-                micoApplication = replaceMicoServiceInApplication(micoApplication, associatedMicService, micoServiceFromParameter);
-                applicationRepository.save(micoApplication);
-            }
+            // of the application need to be updated ...
+            application.getServices().add(service);
+            application.getServiceDeploymentInfos().add(serviceDeploymentInfo);
+            // ... before the application can be saved.
+            applicationRepository.save(application);
+        } else if (micoServicesWithIdenticalShortName.size() == 1) {
```
Comment (wagnerdk) [inline mico-core/src/main/java/io/github/ust/mico/core/resource/ApplicationResource.java]: Why did you remove the constant? It explained the magic number.

File: mico-core/src/main/java/io/github/ust/mico/core/resource/ApplicationResource.java [hunk 3]
```diff
@@ -256,82 +256,72 @@
...
+        	
+        	// ... but only replace if the new version is different from the current version
+        	if (currentService.getVersion().equals(serviceVersion)) {
+        		log.info("Application '{}' '{}' already contains service '{}' in version '{}'. Service will not be added to the application.",
+                    applicationShortName, applicationVersion, serviceShortName, serviceVersion);
+        	} else {
+        		log.info("Replace service '{}' '{}' in application '{}' '{}' with new version '{}'.",
+                    serviceShortName, currentService.getVersion(), applicationShortName, applicationVersion, serviceVersion);
+        		// Replace service in list of services in application
+        		application.getServices().remove(currentService);
```
Comment (wagnerdk) [inline mico-core/src/main/java/io/github/ust/mico/core/resource/ApplicationResource.java]: why did you move this and the next few lines here from a separate method?
